app/assets/javascripts/discourse/tests/integration/components/site-header-test.js

```diff
@@ -50,6 +50,33 @@ module("Integration | Component | site-header", function (hooks) {
     await click("header.d-header");
   });
 
+  test("rerenders when all_unread_notifications or unseen_reviewable_count change", async function (assert) {
```

Comment:
Given that this is an integration test, `all_unread_notifications` and `unseen_reviewable_count` are internal implementation details so the test description should not be describing something that is internal. Rather, we want to describe the behaviour. For example, 

```
test("displaying unread and reviewable notifications count when user receives notifications and reviewables")
```